File: Materia/04-Exercicios Diagrama/Exer_01/Que_01_cpp/OrdemServico.cpp
#include <string>
using std::string;

#include "OrdemServico.h"
#include "Funcionario.h"
#include "Veiculo.h"
#include "Servico.h"

double OrdemServico::getValorTotalOrdem(){
	return 0.0;
}

void OrdemServico::RemoveFuncionario(Funcionario f){
	listaFuncionario.push_back( f );
}

void OrdemServico::addFuncionario(Funcionario f){

}

string OrdemServico::getDataConclusao() const
{
    return dataConclusao;
}

string OrdemServico::getDataExp() const
{
    return dataExp;
}

long OrdemServico::getNumero() const
{
    return numero;
}

void OrdemServico::setDataConclusao(string dataConclusao)
{
    this->dataConclusao = dataConclusao;
}

void OrdemServico::setDataExp(string dataExp)
{
    this->dataExp = dataExp;
}

void OrdemServico::setNumero(long  numero)
{
    this->numero = numero;
}

Cliente OrdemServico::getCliente() const
{
    return cliente;
}

void OrdemServico::setCliente(Cliente cliente)
{
    this->cliente = cliente;
}

Veiculo OrdemServico::getVeiculo() const
{
    return veiculo;
}

void OrdemServico::setVeiculo(Veiculo veiculo)
{
    this->veiculo = veiculo;
}

Servico OrdemServico::getServico() const
{
    return servico;
}

void OrdemServico::setServico(Servico servico)
{
    this->servico = servico;
}

File: Materia/04-Exercicios Diagrama/Exer_01/Que_01_Java/OrdemServico.java
package model;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class OrdemServico {

    private long numero;
    private Date dataExp;
    private Date dataConclusao;
    
    private List<Funcionario> funcionarios = new ArrayList<>();
    private Cliente cliente;
    private List<Veiculo> veiculo;
    
    private List<Servico> servicos = new ArrayList<>();
    
    public double getValorTotalOrdem(){
        return 0.0;
    }
    
    public void addFuncionario( Funcionario f ){
        funcionarios.add(f);
    }
    
    public void removeFuncionario( Funcionario f ){
        funcionarios.remove(f);
    }
    
    public void addServico( Servico s ){
        servicos.add(s);
    }
    
    public void removeServico( Servico s ){
        servicos.remove(s);
    }

    public long getNumero() {
        return numero;
    }

    public void setNumero(long numero) {
        this.numero = numero;
    }

    public Date getDataExp() {
        return dataExp;
    }

    public void setDataExp(Date dataExp) {
        this.dataExp = dataExp;
    }

    public Date getDataConclusao() {
        return dataConclusao;
    }

    public void setDataConclusao(Date dataConclusao) {
        this.dataConclusao = dataConclusao;
    }
}
